pkg/registry/apis/provisioning/repository/git/repository.go

```diff
@@ -400,11 +401,15 @@ func (r *gitRepository) Write(ctx context.Context, path string, ref string, data
 	}
 
 	ctx, _ = r.logger(ctx, ref)
-	_, err := r.Read(ctx, path, ref)
+	info, err := r.Read(ctx, path, ref)
 	if err != nil && !(errors.Is(err, repository.ErrFileNotFound)) {
 		return fmt.Errorf("check if file exists before writing: %w", err)
 	}
 	if err == nil {
+		// If the value already exists and is the same, we don't need to do anything
+		if bytes.Equal(info.Data, data) {
```

Comment:
we can use hash comparison which will be less expensive. I will fix it in the other PR.